lib/compiler-singlepass/src/emitter_arm64.rs

```diff
@@ -1336,13 +1342,13 @@ impl EmitterARM64 for Assembler {
                 let src1 = src1.into_index() as u32;
                 let src2 = src2.into_index() as u32;
                 let dst = dst.into_index() as u32;
-                dynasm!(self ; add X(dst), X(src1), X(src2));
+                dynasm!(self ; add X(dst), X(src1), X(src2), UXTX);
```

Comment:
Just checking that I understand correctly:

UXTX forces a specific instruction encoding, and thus prevents us from special-casing the instruction on the value, correct?

Reply:
Correct. the `sub reg, reg, reg` opcode as 2 form. The "extended register" and the "shifted register'. Only the extended register can be used with XSP, the shifted register will use XZR. The UXTX for the use of the "extended register" form, but in practice does nothing special (UXTX is Zero Extend from 64bits to 64bits).

Reply:
:+1: 

Thanks for the explanation, I'm enjoying brushing up on my Assembly knowledge.